spec/models/user_spec.rb

```diff
@@ -44,4 +44,20 @@
       end
     end
   end
+
+  describe '#has_never_logged_in?' do
+    context 'user has never logged in' do
+      it 'returns true' do
+        user = User.new(logged_in_at: nil)
```

Comment:
We could use `FactoryBot` to build the `User` object. I used `User` to keep it relatively simple.